conan/tools/cmake/cmakedeps.py

```diff
@@ -619,83 +583,88 @@ def content(self):
             conan_package_library_targets,
         ])
 
-        for pkg_name, cpp_info in self._conanfile.deps_cpp_info.dependencies:
-            self._validate_components(cpp_info)
-            pkg_filename = self._get_filename(cpp_info, pkg_name)
-            pkg_findname = self._get_name(cpp_info, pkg_name)
-            pkg_version = cpp_info.version
-
-            public_deps = self.get_public_deps(cpp_info)
-            deps_names = []
-            for it in public_deps:
-                name = "{}::{}".format(*self._get_require_name(*it))
-                if name not in deps_names:
-                    deps_names.append(name)
-            deps_names = ';'.join(deps_names)
-            pkg_public_deps_filenames = [self._get_filename(self._conanfile.deps_cpp_info[it[0]],
-                                                            pkg_name)
-                                         for it in public_deps]
-            config_version = self.config_version_template.format(version=pkg_version)
+        host_requires = {r.ref.name: r for r in
+                         self._conanfile.dependencies.transitive_host_requires}
+        # FIXME: convert to NewCppInfo into the conanfile_interface.py
+        #  "cpp_info" getter for all the new generators
+        for req in host_requires.values():
+            req._cpp_info = NewCppInfo.from_old_cppinfo(req._conanfile.cpp_info)
```

Comment:
It shouldn't be the ``CMakeDeps`` the one creating and assigning a private ``req._cpp_info``. I think maybe this responsibility can be moved to ``conanfile.py``?